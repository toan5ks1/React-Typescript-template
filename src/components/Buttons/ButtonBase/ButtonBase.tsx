import { Button as MaterialButton, ButtonProps as MaterialButtonProps } from "@mui/material";
import CircularProgressBase from "src/components/CicularProgress/CicularProgressBase";

export interface ButtonBaseProps extends Omit<MaterialButtonProps, "color"> {
    color?: MaterialButtonProps["color"] | "error";
    isLoading?: boolean;
}

const ButtonBase = ({
    color = "default", // set as default to consist with V4 color prop
    variant,
    children,
    className,
    isLoading,
    startIcon,
    sx = [],
    ...rest
}: ButtonBaseProps) => {
    const renderedStartIcon = isLoading ? <CircularProgressBase sizeVariant="button" /> : startIcon;

    return (
        <MaterialButton
            className={className}
            sx={[
                color === "error"
                    ? (theme) => {
                          if (variant === "contained") {
                              return {
                                  backgroundColor: theme.palette.error.main,
                                  color: theme.palette.error.contrastText,
                                  "&:hover": {
                                      backgroundColor: theme.palette.error.dark,
                                  },
                              };
                          } else {
                              return {
                                  color: theme.palette.error.main,
                              };
                          }
                      }
                    : {},
                ...(Array.isArray(sx) ? sx : [sx]),
            ]}
            variant={variant}
            color={color !== "error" ? color : "default"}
            startIcon={renderedStartIcon}
            {...rest}
        >
            {children}
        </MaterialButton>
    );
};

export default ButtonBase;
