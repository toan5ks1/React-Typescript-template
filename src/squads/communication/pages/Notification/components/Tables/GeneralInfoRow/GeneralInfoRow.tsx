import { ERPModules } from "src/common/constants/enum";

import { Grid, GridSize } from "@mui/material";
import TypographyTextSecondary from "src/components/Typographys/TypographyTextSecondary";

import useResourceTranslate from "src/squads/communication/hooks/useResourceTranslate";

interface GeneralInfoRowProps {
    label: string;
    children?: React.ReactNode;
    xsLabel?: GridSize;
    xsValue?: GridSize;
}

const GeneralInfoRow = ({ label, children, xsLabel = 1, xsValue = 11 }: GeneralInfoRowProps) => {
    const tNotification = useResourceTranslate(ERPModules.NOTIFICATIONS);

    return (
        <Grid container item>
            <Grid item xs={xsLabel}>
                <TypographyTextSecondary
                    data-testid="GeneralInfoRow__label"
                    variant="caption"
                >{`${tNotification(label)}`}</TypographyTextSecondary>
            </Grid>
            <Grid item xs={xsValue}>
                {children}
            </Grid>
        </Grid>
    );
};

export default GeneralInfoRow;
