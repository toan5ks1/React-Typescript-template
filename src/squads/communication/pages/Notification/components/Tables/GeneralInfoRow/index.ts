export { default } from "./GeneralInfoRow";
export * from "./GeneralInfoRow";
