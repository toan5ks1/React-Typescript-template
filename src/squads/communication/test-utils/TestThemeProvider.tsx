import { PropsWithChildren } from "react";

import { getThemeWithMuiV5 } from "src/styles";

import { ThemeProvider as ThemeProviderV5 } from "@mui/material/styles";

//currently we will expect the style come from manabie first, in the future we will do snapshot separately for each style options
const themeV5 = getThemeWithMuiV5();

const TestThemeProvider = ({ children }: PropsWithChildren<{}>) => {
    return <ThemeProviderV5 theme={themeV5}>{children}</ThemeProviderV5>;
};

export default TestThemeProvider;
