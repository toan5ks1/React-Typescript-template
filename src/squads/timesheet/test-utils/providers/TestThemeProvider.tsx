import { PropsWithChildren } from "react";

import { getThemeWithMuiV5 } from "src/styles";

import { ThemeProvider } from "@mui/material/styles";

//currently we will expect the style come from manabie first, in the future we will do snapshot separately for each style options
const theme = getThemeWithMuiV5();

const TestThemeProvider = ({ children }: PropsWithChildren<{}>) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default TestThemeProvider;
