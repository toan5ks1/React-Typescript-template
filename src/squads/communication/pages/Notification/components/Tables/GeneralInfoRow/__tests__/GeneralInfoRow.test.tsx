import ButtonBase from "src/components/Buttons/ButtonBase";

import GeneralInfoRow from "../GeneralInfoRow";

import { render, RenderResult } from "@testing-library/react";
import TestThemeProvider from "src/squads/communication/test-utils/TestThemeProvider";

describe("<GeneralInfoRow />", () => {
    let wrapper: RenderResult;

    beforeEach(() => {
        wrapper = render(
            <TestThemeProvider>
                <GeneralInfoRow label="Test" xsLabel={2} xsValue={10}>
                    <ButtonBase data-testid="ButtonBase__Test">Test</ButtonBase>
                </GeneralInfoRow>
            </TestThemeProvider>
        );
    });
    it("should snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render correct value", () => {
        expect(wrapper.getByTestId("GeneralInfoRow__label")).toHaveTextContent(
            "resources.notifications.Test"
        );
        expect(wrapper.getByTestId("ButtonBase__Test")).toHaveTextContent("Test");
    });
});
