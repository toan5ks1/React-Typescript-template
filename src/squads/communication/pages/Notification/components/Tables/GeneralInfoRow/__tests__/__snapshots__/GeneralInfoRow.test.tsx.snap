// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeneralInfoRow /> should snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 16.666667%;
  -ms-flex-preferred-size: 16.666667%;
  flex-basis: 16.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 16.666667%;
}

@media (min-width:600px) {
  .emotion-1 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

@media (min-width:960px) {
  .emotion-1 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

@media (min-width:1440px) {
  .emotion-1 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

@media (min-width:1920px) {
  .emotion-1 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

.emotion-2 {
  margin: 0;
  font-size: 0.75rem;
  font-weight: 400;
  letter-spacing: 0.4px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.66;
  color: #757575;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 83.333333%;
  -ms-flex-preferred-size: 83.333333%;
  flex-basis: 83.333333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 83.333333%;
}

@media (min-width:600px) {
  .emotion-3 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

@media (min-width:960px) {
  .emotion-3 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

@media (min-width:1440px) {
  .emotion-3 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

@media (min-width:1920px) {
  .emotion-3 {
    -webkit-flex-basis: 83.333333%;
    -ms-flex-preferred-size: 83.333333%;
    flex-basis: 83.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 83.333333%;
  }
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #E0E0E0;
  border-radius: 4px;
  color: #212121;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(224, 224, 224, 0.04);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-disabled {
  color: #BDBDBD;
}

.emotion-4:hover {
  background-color: rgba(33, 33, 33, 0.04);
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item emotion-0"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 emotion-1"
    >
      <span
        class="MuiTypography-root MuiTypography-caption emotion-2"
        data-testid="GeneralInfoRow__label"
      >
        resources.notifications.Test
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 emotion-3"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textDefault MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  emotion-4"
        data-testid="ButtonBase__Test"
        tabindex="0"
        type="button"
      >
        Test
        <span
          class="MuiTouchRipple-root emotion-5"
        />
      </button>
    </div>
  </div>
</div>
`;
